### ISSUE TITLE
[Feature] User can log out

### ISSUE THREAD

**Issue by blyscuit** [pt-hung1311/Nimble#3]
## Issue

A logout feature would help testing the application smoother. Also leverage the token system in the application.

## Expected

- Home screen should show a profile button on the top right as per design.
    - Profile button can show a placeholder image
    - Or use data from the API to show the image.
- Tapping the profile button should show a profile screen.
    - The profile screen does not need to be partially shown like in design, but can cover the whole screen for ease of implementation.
- Profile screen should should a log out button like design.
- Tapping the log out button should *log out* the user.
- Use SwiftUI for any part of the UI in this issue, if not possible, please comment on why.

**Comment by pt-hung1311** [comment]
The requirement of this project is need support iOS from version 10.0. So we can't use SwiftUI in this project. And from the limitation of the architecture. I need to update some stuff to adapt to SwiftUI

**Comment by blyscuit** [comment]
@pt-hung1311 How familiar are you with programmatic UI? What decision did you make to not use programmatic UI in this assignment? Please explain, if you were to use programmatic UI to create HomeViewController, how would you proceed?

**Comment by pt-hung1311** [comment]
Yes, I'm familiar with programmatic UI. At the current company, the UI will create by programmatic with SnapKit. 
The components if I use programmatic will follow:
1. topStackView: UIStackView(horizontal)
    - dateStackView(vertical): Including 2 UILabel (dateLabel, todayLabel)
    - UIImageView
2. UICollectionView with paging config
3. bottomStackView:
    - pageControl
    - detailStackView(horizontal):
        - stackView(vertical): titleLabel, descriptionLabel
        - actionStackView(vertical): UIView, actionButton, UIView (add UIView here to ensure the main detailStackView does not change the size of the actionButton)
